packages/entity/src/EntityMutator.ts

```diff
@@ -517,6 +514,14 @@ export class UpdateMutator<
 
     return result(updatedEntity);
   }
+
+  private ensureStableIDField(updatedFields: Partial<TFields>): void {
+    const originalId = this.originalEntity.getID();
+    const idField = this.entityConfiguration.idField;
+    if (idField in updatedFields && originalId !== updatedFields[idField]) {
```

Comment:
In practice this won't matter, but if idField were "constructor" or "toString", we don't want to confuse those with Object.prototype.constructor, etc.

```suggestion
    if (updatedFields.hasOwnProperty(idField) && originalId !== updatedFields[idField]) {
```

It probably wouldn't be a bad idea to ban fields named any of the following:
```
Object.prototype.__proto__
Object.prototype.constructor
Object.prototype.hasOwnProperty
Object.prototype.isPrototypeOf
Object.prototype.propertyIsEnumerable
Object.prototype.toLocaleString
Object.prototype.toString
Object.prototype.valueOf
```

Technically it is possible to write code in a way that allows for these fields (use `hasOwnProperty()` instead of `in`, including `for ... in`) but "constructor" is the only field I could see someone wanting to use and they should just choose another name and move on.

Reply:
Will do the second portion of disallowing prototype names as field names in a follow-up PR.

Reply:
https://github.com/expo/entity/pull/236